snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/ecommerce/entities/Product.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2015-2023 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.ecommerce.entities
+
+/**
+ * Used for Ecommerce events.
+ */
+data class Product(
+    /**
+     * The SKU or product ID
+     */
+    val id: String,
+
+    /**
+     * The name or title of the product
+     */
+    val name: String? = null,
```

Comment:
Maybe we should keep the properties mutable (`var`) so that users can initialize them also later instead of just in the constructor if they prefer to?